src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
```

Comment:
Any magic numbers should be constants. It's good that you've defined `DEFAULT_CAPACITY` as a constant, but the value for increasing the array size (`HALF_CAPACITY`) is also a magic number and should be a constant with a clear name explaining its purpose.